### PR TITLE
Create a new endpoint in ServerConfigurationController which will handle a form for submitting both baseUrl and port

### PR DESCRIPTION
**Pull request by ptirador**
Related issue: #661 

New method created:
Name:`setServerSettings(ServerSettingsForm, BindingResult, String)`
Endpoint: `/serverSettings`

The proper tests cases have been modified and new cases have been added.

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/ServerConfigurationController.java [hunk 1]
```diff
@@ -148,6 +155,29 @@ public ResponseEntity getPort(@RequestHeader(HttpHeaders.ACCEPT) String accept)
...
+    {
+        if (bindingResult.hasErrors())
+        {
+            throw new RequestBodyValidationException(FAILED_SAVE_SERVER_SETTINGS, bindingResult);
+        }
+
+        configurationManagementService.setBaseUrl(serverSettingsForm.getBaseUrl());
+        configurationManagementService.setPort(serverSettingsForm.getPort());
+
+        return ResponseEntity.ok(getResponseEntityBody(SUCCESSFUL_SAVE_SERVER_SETTINGS, acceptHeader));
```
Comment (steve-todorov) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/ServerConfigurationController.java]: This should actually be `return getSuccessfulResponseEntity(SUCCESSFUL_SAVE_SERVER_SETTINGS, acceptHeader)` :)

**Comment by steve-todorov** [comment]
Thanks for fixing this @ptirador! I'll merge it once you have a chance to look at my commend above. :)

**Comment by ptirador** [comment]
@steve-todorov Done! :)

**Comment by steve-todorov** [comment]
Thanks! :)

**Pull request merged**